src/commands/openapi/upload.ts

```diff
@@ -0,0 +1,214 @@
+      command: '<%= config.bin %> <%= command.id %> --version=1.0.0 openapi.json',
+    },
+    {
+      description:
+        'You can also pass in a file in a subdirectory (we recommend always running the CLI from the root of your repository):',
+      command: '<%= config.bin %> <%= command.id %> --version=v1.0.0 example-directory/petstore.json',
+    },
+    {
+      description: 'You can also pass in a URL:',
+      command: '<%= config.bin %> <%= command.id %> --version=1.0.0 https://example.com/openapi.json',
+    },
+    {
+      description:
+        'If you specify your ReadMe project version in the `info.version` field in your OpenAPI definition, you can use that:',
+      command: '<%= config.bin %> <%= command.id %> --useSpecVersion https://example.com/openapi.json',
+    },
+  ];
+
+  /**
+   * Poll the ReadMe API until the upload is complete.
+   */
+  private async pollAPIUntilUploadIsComplete(slug: string, headers: Headers) {
+    let count = 0;
+    let status = 'pending';
+
+    while (status === 'pending' && count < 10) {
+      // eslint-disable-next-line no-await-in-loop, no-loop-func
+      await new Promise(resolve => {
+        // exponential backoff — wait 1s, 2s, 4s, 8s, 16s, 32s, 30s, 30s, 30s, 30s, etc.
+        setTimeout(resolve, Math.min(isTest() ? 1 : 1000 * 2 ** count, 30000));
+      });
+      this.debug(`polling API for status of ${slug}, count is ${count}`);
+      // eslint-disable-next-line no-await-in-loop
+      const response = await this.readmeAPIFetch(slug, { headers }).then(res => this.handleAPIRes(res));
+      status = response?.data?.upload?.status;
+      count += 1;
+    }
+
+    if (status === 'pending') {
+      throw new Error('Sorry, this upload timed out. Please try again later.');
+    }
+
+    return status;
+  }
+
+  async run() {
+    const { spec } = this.args;
+
+    const { preparedSpec, specFileType, specPath, specVersion } = await prepareOas(spec, 'openapi');
+
+    const version = this.flags.useSpecVersion ? specVersion : this.flags.version;
+
+    let filename = specFileType === 'url' ? nodePath.basename(specPath) : slugify.default(specPath);
+
+    if (this.flags.slug) {
+      const fileExtension = nodePath.extname(filename);
+      const slugExtension = nodePath.extname(this.flags.slug);
+      if (slugExtension && (!['.json', '.yaml', '.yml'].includes(slugExtension) || fileExtension !== slugExtension)) {
+        throw new Error(
+          'Please provide a valid file extension that matches the extension on the file you provided. Must be `.json`, `.yaml`, or `.yml`.',
+        );
+      }
+
+      // the API expects a file extension, so keep it if it's there, add it if it's not
+      filename = `${this.flags.slug.replace(slugExtension, '')}${fileExtension}`;
+    }
+
+    const headers = new Headers({ authorization: `Bearer ${this.flags.key}` });
+
+    const existingAPIDefinitions = await this.readmeAPIFetch(`/versions/${version}/apis`, { headers }).then(res =>
```

Comment:
so the reason we didn't do that in API v1 (and hence why i didn't do it here) is b/c we occasionally interacted with endpoints that didn't have auth (namely the API registry `POST` endpoint) and i figured this would future-proof things a bit. we can probably add an optional arg though!

Reply:
cool with it as-is, was just curious